sklearn/cluster/hierarchical.py

```diff
@@ -109,7 +111,7 @@ def _single_linkage_tree(connectivity, n_samples, n_nodes, n_clusters,
 
     # Convert edge list into standard hierarchical clustering format
     single_linkage_tree = _hierarchical._single_linkage_label(mst_array)
-    children_ = single_linkage_tree[:, :2].astype(np.int)
+    children_ = single_linkage_tree[:, :2].astype(np.int, copy=False)
```

Comment:
Actually, here I think that it's beneficial to have a copy. It's a somewhat subtle reason:

We are taking a view of the data that is smaller than the original data, via the slicing. This view references the original data. If no copy is made, the original data cannot be garbage collected, and memory is wasted.

Reply:
to fix or not?

Reply:
Will fix tomorrow...

Reply:
Agreed, removed the `copy=False` here.